### PR TITLE
imprv: S3C-3475 add s3 actions in logs

### PR DESCRIPTION
**Pull request by dora-korpar**
Adds fields needed for audit logs

### CODE REVIEW

File: lib/api/api.js [hunk 1]
```diff
@@ -55,6 +55,11 @@ auth.setHandler(vault);
 /* eslint-disable no-param-reassign */
 const api = {
     callApiMethod(apiMethod, request, response, log, callback) {
+        log.addDefaultFields({
+            service: 's3',
+            action: apiMethod,
```
Comment (scality-gdoumergue) [inline lib/api/api.js]: unfortunately apiMethod != s3 action. For example, when the S3 action is [GetObject](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html), the apiMethod is objectGet.

Can't we take advantage of the actionMap object from apiUtils/authorization/permissionChecks.js ?

Comment (dora-korpar) [reply lib/api/api.js]: Ah yeah, we reverse the action names in the apiMethod. The actionMap is a good idea. Thanks!

File: lib/api/api.js [hunk 2]
```diff
@@ -50,11 +50,18 @@ const writeContinue = require('../utilities/writeContinue');
...
+const actionMap = policies.actionMaps.allActionMap;
+
 auth.setHandler(vault);
 
 /* eslint-disable no-param-reassign */
 const api = {
     callApiMethod(apiMethod, request, response, log, callback) {
+        log.addDefaultFields({
+            service: 's3',
+            action: actionMap[apiMethod],
```
Comment (scality-gdoumergue) [inline lib/api/api.js:62]: What happens if the apiMethod is not in the actionMap keys?

Comment (dora-korpar) [reply lib/api/api.js:62]: @scality-gdoumergue It defaults to the most generic api name for the http method. For example, a 'DeleteBucketEncryption' call gets displayed as 's3:DeleteBucket' or a 'GetObjectTorrent' call is displayed as 's3:GetObject'. Is that acceptable behavior, or would it be better to leave out the action if it's not supported?

**Review by tmacro — APPROVED** [review]

**Review by terrydervaux — APPROVED** [review]
LGTM

s3:CreateBucket
`{"name":"S3","bucketName":"test-logs","bytesReceived":0,"bodyLength":0,"service":"s3","action":"s3:CreateBucket","accountDisplayName":"test","accountName":"test","clientIP":"::ffff:127.0.0.1","clientPort":62355,"httpCode":200,"httpMethod":"PUT","httpURL":"/test-logs","time":1605739823642,"req_id":"7c6d60b6894fb00bf6cb","elapsed_ms":24.203538,"level":"info","message":"responded to request","hostname":"Terrys-MacBook-Pro.local","pid":95832}`

s3:PutObject
`{"name":"S3","bucketName":"test-logs","objectKey":"test-object-01","bytesReceived":0,"bodyLength":0,"service":"s3","action":"s3:PutObject","accountDisplayName":"test","accountName":"test","clientIP":"::ffff:127.0.0.1","clientPort":62409,"httpCode":200,"httpMethod":"PUT","httpURL":"/test-logs/test-object-01","contentLength":0,"time":1605740095864,"req_id":"f94f86525a49877c3729","elapsed_ms":9.555145,"level":"info","message":"responded to request","hostname":"Terrys-MacBook-Pro.local","pid":95832}`

**Pull request merged**